### ISSUE TITLE
Get rid of console on Windows

### ISSUE THREAD

**Issue by fkaa** [iamgreaser/iceball#234]
Currently we don't do anything to the linker flags, so we get both a console and the Iceball window when launching on windows. If we were to get rid of the console it'd be nice to have something like #130 implemented first.

To get rid of the console we must both link to `SDL2main` and only specify the `WINDOWS` module in the linker arguments like this:
```CMake
set(CMAKE_EXE_LINKER_FLAGS  "${CMAKE_EXE_LINKER_FLAGS} -mwindows")
```

**Comment by rakiru** [comment]
With these args, does it still output to the console if run from it?

**Comment by fkaa** [comment]
@rakiru yes, it just hides it when double click on the .exe for example. Unfortunately Visual Studio doesn't have a way to show the output, but I can probably add a option for it to use while devving.

**Comment by rakiru** [comment]
`#ifdef _DEBUG` or w/e I guess.

**Comment by iamgreaser** [comment]
Considering that the window disappears when any code crashes and you have to run from the commandline to see the error anyway, I'm OK with this.
